cast-align werror on mips

mokutil.c: In function 'sb_state':
mokutil.c:1822:21: error: cast increases required alignment of target type [-Werror=cast-align]
   state = (int32_t)*(uint32_t *)data;
                     ^
mokutil.c:1824:21: error: cast increases required alignment of target type [-Werror=cast-align]
   state = (int32_t)*(uint16_t *)data;
                     ^
cc1: all warnings being treated as errors

Getting the same for Alpinelinux package (only 32bits)
```
mokutil.c: In function 'generate_pw_hash':
mokutil.c:2038:16: error: comparison of integer expressions of different signedness: 'unsigned int' and 'int' [-Werror=sign-compare]
 2038 |  if (salt_size > settings_len - (next - settings)) {
      |                ^
mv -f .deps/mokutil-password-crypt.Tpo .deps/mokutil-password-crypt.Po
```

--- a/src/mokutil.c
+++ b/src/mokutil.c
@@ -2011,7 +2011,7 @@ generate_pw_hash (const char *input_pw)
 	char *password = NULL;
 	char *crypt_string;
 	const char *prefix;
-	int settings_len = sizeof (settings) - 2;
+	size_t settings_len = sizeof (settings) - 2;
 	unsigned int pw_len, salt_size;
 
 	if (input_pw) {

